src/lib/requests.ts

```diff
@@ -0,0 +1,20 @@
+export interface DeviceResponse {
+  data: {
+    devices: {
```

Comment:
Awesome. Your getting into TS. :) 

I suggest splitting interfaces so You have the DeviceResponse and devices is from the Type Device []

e.g.
```typescript
interface DeviceResponse{
	data: {
	devices: Device[];
	}
}

interface Device {
	id: string;
	// …
}
```

Reply:
Just a question. Does the API return devices as Object or as Array of Device?

Reply:
As an array of devices. I might still be messing up some type declarations, let me know then ;) But also, there will be a few more commits in this PR. I would ask you to take a look later today

Reply:
Sure. I'll give it a test ride later on.